version: '3.7'

services:
  app:
    image: chaosdorf/labello:latest
    environment:
      - PRINTER_HOST=10.1.255.228
    secrets:
      - source: LABELLO_SENTRY_DSN
        target: SENTRY_DSN
    networks:
      - traefik
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        delay: 60s
        max_attempts: 5
      resources:
        limits:
          cpus: '0.50'
          memory: 128M
        reservations:
          cpus: '0.25'
          memory: 32M
      labels:
        - traefik.docker.network=traefik_net
        - traefik.http.services.labello.loadbalancer.server.port=8000
        - "traefik.http.routers.labello-http.rule=Host(`labello.chaosdorf.space`) || Host(`labello`)"
        - traefik.http.routers.labello-http.middlewares=global-headers@file,labello-error@docker
        - traefik.http.routers.labello-http.service=labello@docker
        - "traefik.http.routers.labello-https.rule=Host(`labello.chaosdorf.space`) || Host(`labello`)"
        - traefik.http.routers.labello-https.middlewares=global-headers@file,labello-error@docker
        - traefik.http.routers.labello-https.service=labello@docker
        - traefik.http.routers.labello-https.tls=true
        - traefik.http.routers.labello-https.tls.certresolver=default
        - traefik.http.routers.labello-https.tls.domains[0].main=*.chaosdorf.space
        - traefik.http.middlewares.labello-error.errors.service=error@docker
        - traefik.http.middlewares.labello-error.errors.status=401,403,404,429,503
        - traefik.http.middlewares.labello-error.errors.query=/{status}.html

networks:
  traefik:
    driver: overlay
    external: true
    name: traefik_net

secrets:
  LABELLO_SENTRY_DSN:
    external: true
